librosa/display.py

```diff
@@ -638,20 +667,33 @@ class AdaptiveWaveplot:
     waveshow
     """
 
-    def __init__(self, times, y, steps, envelope, sr=22050, max_samples=11025):
+    def __init__(
+        self,
+        times: np.ndarray,
+        y: np.ndarray,
+        steps: Line2D,
+        envelope: PolyCollection,
+        sr: float = 22050,
+        max_samples: int = 11025
+    ):
         self.times = times
         self.samples = y
         self.steps = steps
         self.envelope = envelope
         self.sr = sr
         self.max_samples = max_samples
-        self.cid = None
-        self.ax = None
+        self.cid: Optional[int] = None
```

Comment:
are callback ids guaranteed to be integer?

Reply:
yes according to docs